packages/indexer/src/indexers/autocrat/autocrat-v1_0-indexer.ts

```diff
@@ -0,0 +1,217 @@
+// 5. And from the vaults insert we can insert token accounts
+// from conditional vaults
+// 6. We then fetch the proposals and insert the markets
+// 7. Once markets are inserted then we can insert the proposal
+
+export const AutocratV1_0DaoIndexer: DaoIndexer = {
+  index: async() => {
+    try {
+      // Fetches all daos from the database
+      const daos: DaoAggregate[] = await indexerReadClient.fetchAllDaos();
+
+      // Check to see if we have database values
+      if(daos == undefined || !daos.length){
+        console.warn(`Database seems to be empty, working off chain exclusively.`);
+      }
+      // Map the database dao keys (TODO: We could argue a simple query is better than leveraging the sdk...)
+      const daoKeys: string[] = daos.flatMap((dao) => dao.daos.flatMap((daoKey) => daoKey.publicKey.toString()))
+
+      // Only working for this version (v1.0)
+      // TODO: Review this so we match on the correct key or it has the f=ing key so we're good
+      const onChainDaos = await rpcReadClient.fetchAllDaos();
+
+      // Check to see if the on chain data returns
+      if(onChainDaos == undefined || !onChainDaos.length){
+        console.error(`No on chain data discovered for program, discontinuing.`);
+        // TODO: Throw but we want to keep cycling
+        return Err({ type: AutocratV1_0DaoIndexerError.GeneralError });
+      }
+      // Map the on chain dao keys
+      const onChainDaoKeys: string[] = onChainDaos.daos.map((dao) => dao.publicKey.toString());
+
+      // Get the keys and compare / exclusion filter
+      const daosForEntry = onChainDaoKeys.map((onChainDaoKey) => {
+        const isFound = !daoKeys.find((dbDaoKey) => onChainDaoKey === dbDaoKey)
+        if(isFound){
+          return onChainDaoKey
+        }
+      }).filter((Boolean));
+      
+      // Filter on the remaining keys we have
+      const onChainDaosToInsert: Dao[] = onChainDaos.daos.filter((dao) => {
+        daosForEntry.includes(dao.publicKey.toString());
+      });
+
+      if(!onChainDaosToInsert.length){
+        console.log(`Nothing to insert`);
+        return Err({ type: AutocratV1_0DaoIndexerError.GeneralError });
+      };
+
+      // Insert into database
+      const insertedDaos = onChainDaosToInsert.map(async (dao) => {
+        if(dao.baseToken.publicKey == null || dao.quoteToken.publicKey == null){
+          console.error("Unable to determine public key for dao tokens");
+          return Err({ type: AutocratV1_0DaoIndexerError.GeneralError });
+        }
+        // Puts the base tokens into the DB before we try to insert the dao
+        await usingDb((db) => 
+          db
+            .insert(schema.tokens)
+            .values({
+              mintAcct: dao.baseToken.publicKey?.toString(),
+              name: dao.baseToken.name,
+              symbol: dao.baseToken.name?.toUpperCase(),
+              supply: 1,
+              decimals: 9, // Need to fetch the token decimals from the sdkkkkkk
```

Comment:
This might already list in dao.baseToken.decimals. I am pretty sure it does actually.

Reply:
It's actually not, if you can believe it... Then again my stuff could be borked

Reply:
that doesn't sound right. If you have the name you should have everything else... that really doesn't sound right unless all that stuff changed a bunch since I implemented it lol